portality/static/widget/fixed_query_src_edges.js

```diff
@@ -0,0 +1,127 @@
+                    query.setQueryString(es.newQueryString(params.queryString));
+                    if(params.hasOwnProperty(`sortFilter`)) {
+                        query.setSortBy(es.newSort(params.sortFilter));
+                    }
+                    if(params.hasOwnProperty(`es_type`)) {
+                        query.addMust(es.newTermFilter({field: "es_type", value: params.es_type}));
+                    }
+                }
+                return query;
+            },
+
+            buildParams: function() {
+                let widget_fv_opts = {};
+
+                // If options have been set using the Export method, use only those
+                if (typeof SEARCH_CONFIGURED_OPTIONS !== 'undefined') {
+                    widget_fv_opts.queryString = SEARCH_CONFIGURED_OPTIONS;
+                    widget_fv_opts.rawQuery = true;
+                }
+                // Otherwise, use options provided.
+                else if (typeof QUERY_OPTIONS !== 'undefined') {
+                    widget_fv_opts.rawQuery = false;
+
+                    if(QUERY_OPTIONS.hasOwnProperty('selector')){
+                        widget_fv_opts.selector = QUERY_OPTIONS.selector;
+                    }
+
+                    widget_fv_opts.initialParams = {};
+                    if (QUERY_OPTIONS.hasOwnProperty('page_size')) {
+                        widget_fv_opts.initialParams.size = QUERY_OPTIONS.page_size;
+                    }
+                    if (QUERY_OPTIONS.hasOwnProperty('page_from')) {
+                        widget_fv_opts.initialParams.from = QUERY_OPTIONS.page_from;
+                    }
+
+                    widget_fv_opts.queryString = {};
+                    if (QUERY_OPTIONS.hasOwnProperty('query_string')) {
+                        widget_fv_opts.queryString.queryString = QUERY_OPTIONS.query_string;
+                    }
+                    if (QUERY_OPTIONS.hasOwnProperty('query_field')) {
+                        widget_fv_opts.queryString.defaultField = QUERY_OPTIONS.query_field;
+                    }
+                    if (QUERY_OPTIONS.hasOwnProperty('search_operator')) {
+                        widget_fv_opts.queryString.defaultOperator = QUERY_OPTIONS.search_operator;
+                    }
+
+                    widget_fv_opts.sortFilter = {};
+                    if (QUERY_OPTIONS.hasOwnProperty('sort_field')) {
+                        widget_fv_opts.sortFilter.field = QUERY_OPTIONS.sort_field;
+                    }
+                    if (QUERY_OPTIONS.hasOwnProperty('sort_direction')) {
+                        widget_fv_opts.sortFilter.sort_direction = QUERY_OPTIONS.sort_direction;
+                    }
+
+                    if (QUERY_OPTIONS.hasOwnProperty('search_type')) {
+                        widget_fv_opts.es_type = QUERY_OPTIONS.search_type;
+                    }
+
+                }
+
+                return widget_fv_opts;
```

Comment:
We talked about getting rid of this intermediate stage, is this just waiting for cleanup?

Reply:
no, it doesn't, The widget_fv_options is now the local variable that puts together the params no matter whether they were provided by `QUERY_OPTIONS` or `SEARCH_CONFIGURED_OPTIONS`. Discussed now on call.